In [ ]:
import pandas as pd




In [25]:
import pandas as pd


df = pd.read_csv("Indian_Kids_Screen_Time.csv")

# Check if there are any missing values
print(df.isnull().values.any())

# count missing values in each column
print(df.isnull().sum())


True
Age                                     0
Gender                                  0
Avg_Daily_Screen_Time_hr                0
Primary_Device                          0
Exceeded_Recommended_Limit              0
Educational_to_Recreational_Ratio       0
Health_Impacts                       3218
Urban_or_Rural                          0
dtype: int64


In [24]:


df = pd.read_csv("Indian_Kids_Screen_Time.csv")

# Check for duplicate rows
duplicates = df.duplicated()

duplicates

0       False
1       False
2       False
3       False
4       False
        ...  
9707    False
9708    False
9709    False
9710    False
9711    False
Length: 9712, dtype: bool

In [21]:
print("First 5 rows:")
df.head()

First 5 rows:


Age  Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
0   14    Male                      3.99     Smartphone   
1   11  Female                      4.61         Laptop   
2   18  Female                      3.73             TV   
3   15  Female                      1.21         Laptop   
4   12  Female                      5.89     Smartphone   

   Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
0                        True                               0.42   
1                        True                               0.30   
2                        True                               0.32   
3                       False                               0.39   
4                        True                               0.49   

           Health_Impacts Urban_or_Rural  
0  Poor Sleep, Eye Strain          Urban  
1              Poor Sleep          Urban  
2              Poor Sleep          Urban  
3                     NaN          Urban  
4     Poor Sleep, Anxiety          Urban

In [22]:
df.dtypes



Age                                    int64
Gender                                object
Avg_Daily_Screen_Time_hr             float64
Primary_Device                        object
Exceeded_Recommended_Limit              bool
Educational_to_Recreational_Ratio    float64
Health_Impacts                        object
Urban_or_Rural                        object
dtype: object

In [23]:


# information about the dataframe
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9712 entries, 0 to 9711
Data columns (total 8 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Age                                9712 non-null   int64  
 1   Gender                             9712 non-null   object 
 2   Avg_Daily_Screen_Time_hr           9712 non-null   float64
 3   Primary_Device                     9712 non-null   object 
 4   Exceeded_Recommended_Limit         9712 non-null   bool   
 5   Educational_to_Recreational_Ratio  9712 non-null   float64
 6   Health_Impacts                     6494 non-null   object 
 7   Urban_or_Rural                     9712 non-null   object 
dtypes: bool(1), float64(2), int64(1), object(4)
memory usage: 540.7+ KB


In [ ]:


# missing value anywhere in the dataset
df.isnull().values.any()
df.isnull().sum()



True
Age                                     0
Gender                                  0
Avg_Daily_Screen_Time_hr                0
Primary_Device                          0
Exceeded_Recommended_Limit              0
Educational_to_Recreational_Ratio       0
Health_Impacts                       3218
Urban_or_Rural                          0
dtype: int64


In [20]:
df[df.isnull().any(axis=1)]


Age  Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
3      15  Female                      1.21         Laptop   
6      17    Male                      2.97             TV   
7      10    Male                      2.74             TV   
19      9  Female                      0.00             TV   
22      9  Female                      1.63         Tablet   
...   ...     ...                       ...            ...   
9695   12  Female                      2.69     Smartphone   
9697   16  Female                      4.74     Smartphone   
9699   10  Female                      2.85             TV   
9701   16    Male                      6.08     Smartphone   
9706    8    Male                      4.84     Smartphone   

      Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
3                          False                               0.39   
6                          False                               0.48   
7                           True                               0.54   
19                         False                               0.42   
22                         False                               0.58   
...                          ...                                ...   
9695                       False                               0.48   
9697                        True                               0.41   
9699                        True                               0.49   
9701                        True                               0.38   
9706                        True                               0.55   

     Health_Impacts Urban_or_Rural  
3               NaN          Urban  
6               NaN          Rural  
7               NaN          Urban  
19              NaN          Urban  
22              NaN          Urban  
...             ...            ...  
9695            NaN          Urban  
9697            NaN          Rural  
9699            NaN          Urban  
9701            NaN          Urban  
9706            NaN          Rural  

[3218 rows x 8 columns]

corelation


In [19]:
#corelation
df.corr(numeric_only=True)



Age  Avg_Daily_Screen_Time_hr  \
Age                                1.000000                  0.118328   
Avg_Daily_Screen_Time_hr           0.118328                  1.000000   
Exceeded_Recommended_Limit         0.159173                  0.664950   
Educational_to_Recreational_Ratio -0.488617                 -0.087552   

                                   Exceeded_Recommended_Limit  \
Age                                                  0.159173   
Avg_Daily_Screen_Time_hr                             0.664950   
Exceeded_Recommended_Limit                           1.000000   
Educational_to_Recreational_Ratio                   -0.126643   

                                   Educational_to_Recreational_Ratio  
Age                                                        -0.488617  
Avg_Daily_Screen_Time_hr                                   -0.087552  
Exceeded_Recommended_Limit                                 -0.126643  
Educational_to_Recreational_Ratio                           1.000000

In [ ]:
correlation = df['Age'].corr(df['Avg_Daily_Screen_Time_hr'])

print("Correlation between Age and Avg_Daily_Screen_Time_hr:", correlation)




Correlation between Age and Avg_Daily_Screen_Time_hr: 0.1183276608281807


In [12]:
print(df.describe())


               Age  Avg_Daily_Screen_Time_hr  \
count  9712.000000               9712.000000   
mean     12.979201                  4.352837   
std       3.162437                  1.718232   
min       8.000000                  0.000000   
25%      10.000000                  3.410000   
50%      13.000000                  4.440000   
75%      16.000000                  5.380000   
max      18.000000                 13.890000   

       Educational_to_Recreational_Ratio  
count                        9712.000000  
mean                            0.427226  
std                             0.073221  
min                             0.300000  
25%                             0.370000  
50%                             0.430000  
75%                             0.480000  
max                             0.600000  


Summary for all columns including categorical

In [14]:
df.describe(include='all')


Age Gender  Avg_Daily_Screen_Time_hr Primary_Device  \
count   9712.000000   9712               9712.000000           9712   
unique          NaN      2                       NaN              4   
top             NaN   Male                       NaN     Smartphone   
freq            NaN   4942                       NaN           4568   
mean      12.979201    NaN                  4.352837            NaN   
std        3.162437    NaN                  1.718232            NaN   
min        8.000000    NaN                  0.000000            NaN   
25%       10.000000    NaN                  3.410000            NaN   
50%       13.000000    NaN                  4.440000            NaN   
75%       16.000000    NaN                  5.380000            NaN   
max       18.000000    NaN                 13.890000            NaN   

       Exceeded_Recommended_Limit  Educational_to_Recreational_Ratio  \
count                        9712                        9712.000000   
unique                          2                                NaN   
top                          True                                NaN   
freq                         8301                                NaN   
mean                          NaN                           0.427226   
std                           NaN                           0.073221   
min                           NaN                           0.300000   
25%                           NaN                           0.370000   
50%                           NaN                           0.430000   
75%                           NaN                           0.480000   
max                           NaN                           0.600000   

       Health_Impacts Urban_or_Rural  
count            6494           9712  
unique             15              2  
top        Poor Sleep          Urban  
freq             2268           6851  
mean              NaN            NaN  
std               NaN            NaN  
min               NaN            NaN  
25%               NaN            NaN  
50%               NaN            NaN  
75%               NaN            NaN  
max               NaN            NaN

See how many are missing vs not missing

In [15]:
df['Health_Impacts'].isnull().value_counts()


Health_Impacts
False    6494
True     3218
Name: count, dtype: int64

percentage of missing values

In [18]:
#percentage of missing values
missing_percentage = df['Health_Impacts'].isnull().mean() * 100
print(f"Percentage of missing values in Health_Impacts: {missing_percentage:.2f}%")



Percentage of missing values in Health_Impacts: 33.13%


Observation from the Dataset

After looking at the dataset, here is what I haveve understood:

Structure of the data
The dataset includes information like Age, Gender, how much time kids spend on screens daily, the type of device they use, and other details such as whether they exceeded recommended screen limits and their educational-to-recreational activity ratio. It also has location information like Urban or Rural areas.

Missing Data
 Health Impacts column. This column has a large number of missing values (3218 entries missing), which is something we need to explore further.



# New Section